service-name-handler.js

```diff
@@ -52,7 +54,10 @@ TChannelServiceNameHandler.prototype.handleLazily = function handleLazily(conn,
             error: res.err
         }));
         // TODO: protocol error instead?
-        conn.sendLazyErrorFrame(reqFrame, 'BadRequest', 'failed to read serviceName');
+        res = self.reqFrame.bodyRW.lazy.readTracing(self.reqFrame);
+        tracing = res.err ? v2.Tracing.emptyTracing : res.value;
+
+        conn.sendLazyErrorFrame(reqFrame.id, tracing, 'BadRequest', 'failed to read serviceName');
```

Comment:
actually, let's have our cake and eat it too:
- add back something like `conn.sendLazyErrorFrameFor(reqFrame, code, mess)`